include/nil/crypto3/zk/snark/arithmetization/plonk/detail/lookup_table_definition.hpp

```diff
@@ -0,0 +1,166 @@
+
+                        // Allocate constant columns
+                        std::vector<plonk_column<FieldType>> constant_columns(constant_columns_ids.size());
+                        std::vector<plonk_column<FieldType>> selector_columns;
+                        for(std::size_t i = 0; i < constant_columns.size(); i++){
+                            constant_columns[i] = plonk_column<FieldType>(usable_rows, FieldType::value_type::zero());
+                        }
+
+                        std::size_t start_row = 1;
+                        std::size_t table_index = 0;
+                        std::vector<plonk_lookup_table<FieldType>> bp_lookup_tables(lookup_table_ids.size());
+                        for( const auto&[k, table]:lookup_tables ){
+                            // Place table into constant_columns.
+                            for( std::size_t i = 0; i < table->get_table().size(); i++ ){
+                                if(constant_columns[i].size() < start_row + table->get_table()[i].size()){
+                                    constant_columns[i].resize(start_row + table->get_table()[i].size());
+                                    if( usable_rows_after < start_row + table->get_table()[i].size() ){
+                                        usable_rows_after = start_row + table->get_table()[i].size();
+                                    }
+                                }
+                                for( std::size_t j = 0; j < table->get_table()[i].size(); j++ ){
+                                    constant_columns[i][start_row + j] = table->get_table()[i][j];
+                                }
+                            }
+
+                            for( const auto &[subtable_name, subtable]:table->subtables ){
+                                // Create selector
+                                plonk_column<FieldType> selector_column(usable_rows_after, FieldType::value_type::zero());
+                                for(std::size_t k = subtable.begin; k <= subtable.end; k++){
+                                    selector_column[start_row + k] = FieldType::value_type::one();
+                                }
+
+                                std::string full_table_name = table->table_name + "/" + subtable_name;
+                                bp_lookup_tables[lookup_table_ids.at(full_table_name) - 1] = plonk_lookup_table<FieldType>(subtable.column_indices.size(), cur_selector_id);
+                                std::vector<plonk_variable<typename FieldType::value_type>> option;
+                                for( const auto &column_index:subtable.column_indices ){
+                                    plonk_variable<typename FieldType::value_type> var(
+                                        constant_columns_ids[column_index], 0, false, plonk_variable<typename FieldType::value_type>::column_type::constant
+                                    );
+                                    option.push_back(var);

```

Comment:
Consider 

option.emplace_back(constant_columns_ids[column_index], 0, false, plonk_variable<typename FieldType::value_type>::column_type::constant);

It will probably work slightly faster.